src/controller/InvokeInteraction.h

```diff
@@ -90,6 +90,69 @@ InvokeCommandRequest(Messaging::ExchangeManager * aExchangeMgr, const SessionHan
     return CHIP_NO_ERROR;
 }
 
+/*
+ * A typed group command invocation function that takes as input a cluster-object representation of a command request and
+ * callbacks when completed trought the done callback
+ *
+ * The RequestObjectT is generally expected to be a ClusterName::Commands::CommandName::Type struct, but any object
+ * that can be encoded using the DataModel::Encode machinery and exposes the GetClusterId() and GetCommandId() functions
+ * and a ResponseType type is expected to work.
+ *
+ * Since this sends a group command, no response will be received and all allocated rescources will be cleared before exing this
+ * function
+ */
+template <typename RequestObjectT>
+CHIP_ERROR
+InvokeGroupCommandRequest(Messaging::ExchangeManager * exchangeMgr, chip::FabricIndex fabric, chip::GroupId groupId,
+                          chip::NodeId sourceNodeId, const RequestObjectT & requestCommandData,
+                          typename TypedCommandCallback<typename RequestObjectT::ResponseType>::OnSuccessCallbackType onSuccessCb,
```

Comment:
I will do it but this will cause a divergence between the InvokeInteraction::InvokeGroupCommandRequest and CommandInvoker::InvokeGroupCommand. One will beeing using callbacks and the other won't. 

CommandInvoker::InvokeGroupCommand requires the callbacks because of it's interactions with the yaml tests.

Reply:
`CommandInvoker::InvokeGroupCommand` is app-code, and is free to use callbacks if it makes sense for its use-case. It should be updated to not actually pass in a callback to `CommandSender`, and just manually call the success and on-done callbacks synchronously right after calling `SendGroupCommandRequest`.

Here though, this is core SDK code, and as such, we should aim for an API that makes the most sense for the modality :)

Reply:
I can help do that change

Reply:
ok, in that sense i agree with the no callback logic for the ` InvokeInteraction::InvokeGroupCommandRequest `.

For `CommandInvoker::InvokeGroupCommand`, for the time being, i think we should leave it as it is and focus on finishing everything else that needs finishing.

Reply:
changes are done